httpcore/_synchronization.py

```diff
@@ -18,15 +30,30 @@
     anyio = None  # type: ignore
 
 
-def current_async_library() -> str:
+if sys.version_info >= (3, 11):  # pragma: nocover
+    import asyncio as asyncio_timeout
+
+    anyio_shield = None
+else:  # pragma: nocover
+    import async_timeout as asyncio_timeout
+
+    if anyio is None:  # pragma: nocover
+        raise RuntimeError("Running in Python<3.11 requires anyio")
+    anyio_shield = anyio.CancelScope
+
+
+AsyncBackend = Literal["asyncio", "trio"]
+
+
+def current_async_backend() -> AsyncBackend:
     # Determine if we're running under trio or asyncio.
     # See https://sniffio.readthedocs.io/en/latest/
     try:
         import sniffio
     except ImportError:  # pragma: nocover
-        environment = "asyncio"
+        environment: AsyncBackend = "asyncio"
     else:
-        environment = sniffio.current_async_library()
+        environment = cast(AsyncBackend, sniffio.current_async_library())
 
     if environment not in ("asyncio", "trio"):  # pragma: nocover
```

Comment:
is it worth to renaming `environment` to `backend` now?

Reply:
Done